dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/AnonymisingDataService.java

```diff
@@ -0,0 +1,259 @@
+
+import java.math.BigInteger;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Optional;
+
+import static uk.gov.justice.laa.crime.dces.integration.utils.DateConvertor.convertToXMLGregorianCalendar;
+
+@Slf4j
+@Service
+public class AnonymisingDataService {
+
+    private Faker faker;
+
+    public CONTRIBUTIONS anonymise(CONTRIBUTIONS contribution) {
+
+        SecureRandom secureRandom;
+        try {
+            secureRandom = SecureRandom.getInstance("SHA1PRNG");
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Failed to create SecureRandom instance", e);
+            throw new DcesDrcServiceException(e.getMessage(), e);
+        }
+
+        secureRandom.setSeed(contribution.getMaatId().longValue());
+        faker = new Faker(secureRandom);
+
+        if (hasValue(contribution.getMaatId())) {
+            contribution.setMaatId(BigInteger.valueOf(faker.number().numberBetween(100000, 999999)));
+        }
+        log.info("Anonymising data for contribution with maatId {} and anonymised maat-id: {}", contribution.getMaatId(), contribution.getMaatId());
+        Optional.ofNullable(contribution.getApplicant())
+                .ifPresent(applicant -> contribution.setApplicant(anonymiseApplicant(applicant)));
+        Optional.ofNullable(contribution.getApplication())
+                .ifPresent(application -> contribution.setApplication(anonymiseApplication(application)));
+        Optional.ofNullable(contribution.getCapitalSummary())
+                .ifPresent(capitalSummary -> contribution.setCapitalSummary(anonymiseCapitalSummary(capitalSummary)));
+        Optional.ofNullable(contribution.getEquity())
+                .ifPresent(equity -> contribution.setEquity(anonymiseEquity(equity)));
+
+        return contribution;
+    }
+
+    private CONTRIBUTIONS.Equity anonymiseEquity(CONTRIBUTIONS.Equity equity) {
+        Optional.ofNullable(equity.getPropertyDescriptor()).ifPresent(propertyDescriptor -> {
+            Optional.ofNullable(propertyDescriptor.getThirdPartyList())
+                    .flatMap(thirdPartyList -> Optional.ofNullable(thirdPartyList.getThirdParty()))
+                    .ifPresent(thirdParty -> {
+                        if (hasValue(thirdParty.getName())) {
+                            thirdParty.setName(faker.name().fullName());
+                        }
+                    });
+
+            Optional.ofNullable(propertyDescriptor.getAddress())
+                    .flatMap(address -> Optional.ofNullable(address.getDetail()))
+                    .ifPresent(this::equityAddressDetail);
+        });
+        return equity;
+    }
+
+
+    private CONTRIBUTIONS.CapitalSummary anonymiseCapitalSummary(CONTRIBUTIONS.CapitalSummary capitalSummary) {
+
+        Optional.ofNullable(capitalSummary.getMotorVehicleOwnership()).ifPresent(motorVehicleOwnership -> {
```

Comment:
hi this field doesnt need to be anonymised. Originally I thought it was the Owner Name. But it is actually just a YES or NO field. Therefore please remove the anonymisation from this field